wire-library/wire-moshi-adapter/src/main/java/com/squareup/wire/WireJsonAdapterFactory.kt

```diff
@@ -43,7 +43,25 @@ import java.util.ArrayList
  * compatible with the [protobuf-java-format](https://code.google.com/p/protobuf-java-format/)
  * library.
  */
-class WireJsonAdapterFactory : JsonAdapter.Factory {
+class WireJsonAdapterFactory private constructor(
+  private val typeUrlToAdapter: Map<String, ProtoAdapter<*>>
+) : JsonAdapter.Factory {
+  constructor() : this(mapOf())
+
+  /**
+   * Returns a new WireJsonAdapterFactory that can encode the messages for [adapters] if they're
+   * used with [AnyMessage].
+   */
+  fun plus(vararg adapters: ProtoAdapter<*>) : WireJsonAdapterFactory {
```

Comment:
I think I also want an overload that takes a list? varargs methods don’t compose well

Reply:
went with
```kotlin
  fun plus(adapter: ProtoAdapter<*>): WireJsonAdapterFactory {
// and
  fun plus(adapters: List<ProtoAdapter<*>>): WireJsonAdapterFactory {
```